.ado/jobs/universal.yml

```diff
@@ -5,143 +5,218 @@ parameters:
-            Microsoft.ReactNative.Managed.CodeGen.UnitTests/Microsoft.ReactNative.Managed.CodeGen.UnitTests.dll
-          searchFolder: $(Build.SourcesDirectory)/vnext/target/$(BuildPlatform)/$(BuildConfiguration)
-          runTestsInIsolation: true
-          platform: $(BuildPlatform)
-          configuration: $(BuildConfiguration)
-          publishRunAttachments: true
-          codeCoverageEnabled: true
-          collectDumpOn: onAbortOnly
-          vsTestVersion: latest
-        condition: and(succeeded(), eq(variables.BuildPlatform, 'x64'))
+              - task: PowerShell@2
+                displayName: "Check if this environment meets the development dependencies"
+                inputs:
+                  targetType: filePath
+                  filePath: $(Build.SourcesDirectory)\vnext\Scripts\rnw-dependencies.ps1
+                  arguments: -NoPrompt -Tags buildLab
+                  
+              - template: ../templates/prepare-env.yml
+                parameters:
+                  yarnBuildCmd: build
 
-      - template: ../templates/publish-build-artifacts-for-nuget.yml
-        parameters:
-          artifactName: ReactWindows
-          layoutHeaders: eq('true', variables.LayoutHeaders)
-          contents: |
-            Microsoft.ReactNative\**
-            Microsoft.ReactNative.Managed\**
-            Microsoft.ReactNative.Managed.CodeGen\**
-      
-      - task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
-        displayName: 'Component Governance Detection'
-        inputs:
-          alertWarningLevel: Medium
-          scanType: 'Register'
\ No newline at end of file
+              - task: NuGetCommand@2
+                displayName: NuGet restore
+                inputs:
+                  command: restore
+                  restoreSolution: vnext/Microsoft.ReactNative.sln
+                  feedsToUse: config
+                  nugetConfigPath: $(Build.SourcesDirectory)/vnext/NuGet.config
+                  restoreDirectory: packages/
+                  verbosityRestore: Detailed # Options: quiet, normal, detailed
+                  
+              - task: DownloadPipelineArtifact@1
+                displayName: "Download binaries for testing"
+                inputs:
+                  targetPath: $(Build.SourcesDirectory)/vnext/target/${{ matrix.BuildPlatform }}/${{ matrix.BuildConfiguration }}
+                  artifactName: "Universal-${{ matrix.BuildPlatform }}-${{ matrix.BuildConfiguration }}"
+
+              # The WinDevBuild Agents run on the c-drive and the hosted agetns run on the d-drive.
+              # Some of the output files have the c-drive expanded in the outputs, whcih results in file not found errors when running tests
+              # This script fixes those files to point to the d-drive of the hosted agents.
+              # We unfortunately can't run the tests on the windev build pool because it lacks UI support
```

Comment:
```suggestion
              # The WinDevBuild Agents run on the c-drive and the hosted agents run on the d-drive.
              # Some of the output files have the c-drive expanded in the outputs, which results in file not found errors when running tests
              # This script fixes those files to point to the d-drive of the hosted agents.
              # We unfortunately can't run the tests on the WinDev build pool because it lacks UI support
```

Reply:
thanks!